RadixWallet/Core/FeaturePrelude/FactorSourceCard/FactorSourceCard+DataSource.swift

```diff
@@ -0,0 +1,77 @@
+// MARK: - FactorSourceCardDataSource
+struct FactorSourceCardDataSource {
+	let icon: ImageResource
+	let title: String
+	var subtitle: String?
+	var lastUsedOn: Timestamp?
+	var messages: [Message] = []
+	var accounts: [Account] = []
+	var personas: [Persona] = []
+
+	var hasAccountsOrPersonas: Bool {
+		!accounts.isEmpty || !personas.isEmpty
+	}
+}
+
+// MARK: FactorSourceCardDataSource.Message
+extension FactorSourceCardDataSource {
+	struct Message: Identifiable {
+		var id: String { text }
+
+		let text: String
+		let type: StatusMessageView.ViewType
+	}
+}
+
+extension FactorSourceKind {
+	var icon: ImageResource? {
```

Comment:
this is optional beacause we don't have examples of icon for .trustedContact, .securityQuestions? Or because these should not have not an icon?

Same for the rest, I would put a fatal error instead of returning nil, if the reason is just that we miss icon and so on.